File: app/src/release/java/com/og/emdlc/dagger/AppModule.java
package com.og.emdlc.dagger;

import com.og.emdlc.managers.AccountManager;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Created by olivier.goutay on 3/31/17.
 * Think about the release AppModule (necessary for proguard to avoid mockito in release)
 */
@Module
public class AppModule {

    public AppModule(boolean mockManagers) {
    }

    @Provides
    @Singleton
    AccountManager provideAccountManager() {
        return new AccountManager();
    }
}

File: app/src/release/java/com/og/emdlc/dagger/AppModule.kt
package com.og.emdlc.dagger

import com.og.emdlc.managers.AccountManager
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/**
 * Created by olivier.goutay on 3/31/17.
 * Think about the release AppModule (necessary for proguard to avoid mockito in release)
 */
@Module
class AppModule(mockManagers: Boolean) {

    @Provides
    @Singleton
    internal fun provideAccountManager(): AccountManager {
        return AccountManager()
    }
}
